clusterctl/clusterdeployer/clusterapiservertemplate.go

```diff
@@ -0,0 +1,241 @@
+apiVersion: apiregistration.k8s.io/v1beta1
+kind: APIService
+metadata:
+  name: v1alpha1.cluster.k8s.io
+  labels:
+    api: clusterapi
+    apiserver: "true"
+spec:
+  version: v1alpha1
+  group: cluster.k8s.io
+  groupPriorityMinimum: 2000
+  priority: 200
+  service:
+    name: clusterapi
+    namespace: default
+  versionPriority: 10
+  caBundle: {{ .CABundle }}
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: clusterapi
+  namespace: default
+  labels:
+    api: clusterapi
+    apiserver: "true"
+spec:
+  ports:
+  - port: 443
+    protocol: TCP
+    targetPort: 443
+  selector:
+    api: clusterapi
+    apiserver: "true"
+---
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: clusterapi-apiserver
+  namespace: default
+  labels:
+    api: clusterapi
+    apiserver: "true"
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        api: clusterapi
+        apiserver: "true"
+    spec:
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
+      tolerations:
+      - effect: NoSchedule
+        key: node-role.kubernetes.io/master
+      - key: CriticalAddonsOnly
+        operator: Exists
+      - effect: NoExecute
+        key: node.alpha.kubernetes.io/notReady
+        operator: Exists
+      - effect: NoExecute
+        key: node.alpha.kubernetes.io/unreachable
+        operator: Exists
+      containers:
+      - name: apiserver
+        image: {{ .APIServerImage }}
+        volumeMounts:
+        - name: cluster-apiserver-certs
+          mountPath: /apiserver.local.config/certificates
+          readOnly: true
+        - name: config
+          mountPath: /etc/kubernetes
+        - name: certs
+          mountPath: /etc/ssl/certs
+        command:
+        - "./apiserver"
+        args:
+        - "--etcd-servers=http://etcd-clusterapi-svc:2379"
+        - "--tls-cert-file=/apiserver.local.config/certificates/tls.crt"
+        - "--tls-private-key-file=/apiserver.local.config/certificates/tls.key"
+        - "--audit-log-path=-"
+        - "--audit-log-maxage=0"
+        - "--audit-log-maxbackup=0"
+        - "--authorization-kubeconfig=/etc/kubernetes/admin.conf"
+        - "--kubeconfig=/etc/kubernetes/admin.conf"
+        resources:
+          requests:
+            cpu: 100m
+            memory: 20Mi
+          limits:
+            cpu: 100m
+            memory: 30Mi
+      volumes:
+      - name: cluster-apiserver-certs
+        secret:
+          secretName: cluster-apiserver-certs
+      - name: config
+        hostPath:
+          path: /etc/kubernetes
+      - name: certs
+        hostPath:
+          path: /etc/ssl/certs
+---
+apiVersion: rbac.authorization.k8s.io/
+kind: RoleBinding
+metadata:
+  name: clusterapi
+  namespace: kube-system
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: extension-apiserver-authentication-reader
+subjects:
+- kind: ServiceAccount
+  name: default
+  namespace: default
+---
+apiVersion: apps/v1beta1
```

Comment:
same here

Reply:
This is snapshot-ed to the version called out in the PR description. I want to check it in with a consistent working snapshot so we know what changes it is caught up with and what changes it has yet to catch up on.